static/app/components/dashboards/widgetQueryFields.tsx

```diff
@@ -239,12 +242,12 @@ function WidgetQueryFields({
             <QueryField
               fieldValue={fieldValue}
               fieldOptions={
-                isMetricWidget
-                  ? generateMetricsWidgetFieldOptions()
-                  : generateFieldOptions({organization})
+                isMetricWidget ? fieldOptions : generateFieldOptions({organization})
               }
               onChange={value => handleTopNChangeField(value)}
-              filterPrimaryOptions={filterPrimaryOptions}
+              filterPrimaryOptions={
+                isMetricWidget ? filterMetricsTags : filterPrimaryOptions
```

Comment:
Is the filterMetricsTags name correct? Isn't this for fields?

Reply:
named it poorly!